eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/NoImplInPublicAPI.java

```diff
@@ -0,0 +1,109 @@
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class NoImplInPublicAPI extends AbstractCheck {
+
+    private static final String COM_AZURE = "com.azure";
+    private static final String COM_AZURE_CORE_IMPLEMENTATION = "com.azure.core.implementation";
```

Comment:
You can't just scan the com.azure.core.implementation package - you need to scan for all implementation packages.